Google.PowerShell.IntegrationTests/PubSub/PubSub.Tests.ps1

```diff
@@ -0,0 +1,101 @@
+    $r = Get-Random
+    $script:topicName = "gcps-testing-topic-$r"
+    $script:secondTopicName = "gcps-testing-topic2-$r"
+    $previousCount = (Get-GcpsTopic).Count
+    gcloud beta pubsub topics create $script:topicName 2>$null
+    gcloud beta pubsub topics create $script:secondTopicName 2>$null
+
+    AfterAll {
+        gcloud beta pubsub topics delete $topicName --quiet 2>$null
+        gcloud beta pubsub topics delete $secondTopicName --quiet 2>$null
+    }
+
+    It "should work without any parameters" {
+        $topics = Get-GcpsTopic
+        $topics.Count - $previousCount | Should Be 2
+    }
+
+    It "should work with -Topic parameter" {
+        $firstTopic = Get-GcpsTopic -Topic $topicName
+        $firstTopic.Name | Should BeExactly "projects/$project/topics/$topicName"
+
+        # Should work with an array of topics names.
+        $topics = Get-GcpsTopic -Topic $topicName, $secondTopicName
+        $topics.Count | Should Be 2
+        $topics.Name -contains "projects/$project/topics/$topicName" | Should Be $true
+        $topics.Name -contains "projects/$project/topics/$secondTopicName" | Should Be $true
+    }
+
+    It "should throw an error for non-existent topic" {
+        { Get-GcpsTopic -Topic "non-existent-topic-name" -ErrorAction Stop } | Should Throw "does not exist"
+    }
+}
+
+Describe "New-GcpsTopic" {
+    It "should work" {
+        $r = Get-Random
+        $topicName = "gcp-creating-topic-$r"
+        $secondTopicName = "gcp-creating-topic2-$r"
+        $thirdTopicName = "gcp-creating-topic3-$r"
+
+        try {
+            New-GcpsTopic -Topic $topicName
+            (Get-GcpsTopic -Topic $topicName).Name | Should BeExactly "projects/$project/topics/$topicName"
+
+            # Should work with an array of topics names.
+            New-GcpsTopic -Topic $secondTopicName, $thirdTopicName
+            $topics = Get-GcpsTopic -Topic $secondTopicName, $thirdTopicName
+            $topics.Count | Should Be 2
+            $topics.Name -contains "projects/$project/topics/$secondTopicName" | Should Be $true
+            $topics.Name -contains "projects/$project/topics/$thirdTopicName" | Should Be $true
+        }
+        finally {
+            gcloud beta pubsub topics delete $topicName --quiet 2>$null
+            gcloud beta pubsub topics delete $secondTopicName --quiet 2>$null
+            gcloud beta pubsub topics delete $thirdTopicName --quiet 2>$null
+        }
+    }
+
+    It "should throw an error for topic that is already created" {
+        $r = Get-Random
+        $topicName = "gcp-creating-topic-$r"
+
+        try {
+            New-GcpsTopic -Topic $topicName
+            (Get-GcpsTopic -Topic $topicName).Name | Should BeExactly "projects/$project/topics/$topicName"
+            { New-GcpsTopic -Topic $topicName -ErrorAction Stop } | Should Throw "already exists" 
+        }
+        finally {
+            gcloud beta pubsub topics delete $topicName --quiet 2>$null
+        }
+    }
```

Comment:
It seems like we should have a few more unit tests, just to confirm the behavior is reasonable. (For this and other cmdlets.) For example, what is the error you get if you use an invalid topic name? Do we crash with an obtuse error, or is it reasonable?